.semgrep.yml

```diff
@@ -0,0 +1,60 @@
+rules:
+  - id: translation-no-new-style-formatting
+    patterns:
+      - pattern: $FUNC("$STRING_ID")
+      - metavariable-regex:
+          metavariable: $FUNC
+          regex: '_|gettext|gettext_lazy|ngettext'
+      - metavariable-regex:
+          metavariable: $STRING_ID
+          regex: ".*({(\\d*|[\\w_]*)}).*"
+    message: |
+      Do not use str.format style formatting for translations.
+      Use printf style formatting with named placeholders instead.
+      For example, do `_("Hello %(name)s") % {"name": "Wagtail"}`
+      instead of `_("Hello {name}").format(name="Wagtail")`.
```

Comment:
I think we should really point the developer to a docs page here, hopefully I can get https://github.com/wagtail/wagtail/pull/9431 in and we can have this end docs file as a source of truth here.

The nuances of the various types of string formatting was a bit for me to understand, and there is the temptation to use the 'newer' thing. Having the docs as a place where can explain this a bit better would be useful.

Also, this means adding a docs section also I think as part of this PR.

Reply:
Yes, now that #9431 is merged, I'll rebase my branch tomorrow and piggyback on it 👍